src/server/stats.rs

```diff
@@ -10,14 +10,36 @@ use serde::{Deserialize, Serialize};
 #[derive(Default, Clone, Deserialize, Serialize)]
 pub struct Stats {
     pub accesses: HashMap<String, HashMap<String, u64>>,
+    pub terms: HashMap<String, u64>,
+}
+
+#[derive(Deserialize)]
+struct OldStats {
+    pub accesses: HashMap<String, HashMap<String, u64>>,
 }
 
 impl Stats {
     pub fn read(dir: &Dir) -> Result<Self> {
-        let val = if let Ok(file) = dir.open("stats") {
-            DefaultOptions::new()
+        let val = if let Ok(mut file) = dir.open("stats") {
+            let res = DefaultOptions::new()
                 .with_fixint_encoding()
-                .deserialize_from(BufReader::new(file))?
+                .deserialize_from::<_, Stats>(BufReader::new(&mut file));
+
+            match res {
+                Ok(val) => val,
+                Err(err) => {
+                    let old_val = DefaultOptions::new()
```

Comment:
It would likely make sense to factor out the options into a helper function or ideally even a constant.